### ISSUE TITLE
Plugin doesn't work on cordova-ios 4+

### ISSUE THREAD

**Issue by dpolivy** [quantcast/phonegap-measurement#8]
There have been some changes in `cordova-ios@4`, [described here](https://github.com/apache/cordova-ios/blob/master/guides/API%20changes%20in%204.0.md#cdvpluginh-updated), that make the Quantcast plugin no longer compile.

Specifically, `writeJavascript` and `toSuccessCallbackString` are removed. There's a new way to return plugin results back to JS, as described [here](http://cordova.apache.org/docs/en/dev/guide/platforms/ios/plugin.html).

**Comment by SmithKevin** [comment]
Well dang.   Thanks again for bringing issues to our attention. This change might require a bit more planning on our end as we need to think about keeping backwards compatibility with older versions.  I will put this as work for our next sprint.   I'll keep you posted here on the status.

**Comment by dpolivy** [comment]
I think the "new" way has actually been there for quite some time now -- not sure how far back you really care about, but it may be fine to just update to the new recommended pattern. Cordova has moved along fairly quickly and I don't know how much support is necessary for really old versions, but that's a business decision for you to make.

I may try and hack up a solution in the short term to unblock myself, but I'll keep an eye on this for an official update.

**Comment by SmithKevin** [comment]
Are you using 4.0 or are you using 5+?  I figure if we have to make code changes we might as well make it completely up to date.

**Comment by dpolivy** [comment]
I'm using `cordova@5.4.1`, but `cordova-ios@4.0.1`. They decoupled all the components and now the version numbers are all over the map. It is a bit confusing.

The plugin interface is more determined by the platform version in use (e.g., `cordova-ios`, `cordova-android`) than the tools.

I took a quick look at the [history](https://github.com/apache/cordova-ios/commit/ba7d771306a51c180da7a5541c4738f08a8cac4d), and it seems that `sendPluginResult` has been in there since `cordova-ios@2.2` back in October 2012, so I think it's probably safe to update this without the need for backwards compatibility.

**Comment by dpolivy** [comment]
FWIW, I've worked around this on my own by modifying the source to change the offending lines to something like this:

````
CDVPluginResult* result = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:hash];
[self.commandDelegate sendPluginResult:result callbackId:callbackId];
````

Seems to be fine so far, and this should work as far back as `cordova-ios@2.2`.

**Comment by SmithKevin** [comment]
Thanks @dpolivy.  Good to know.   My changes look the same, but haven't had a chance to test to make sure it works as expect.   I'll push an update as soon as I look into the new issue that just popped up.

**Comment by SmithKevin** [comment]
I have updated both plugins.   This should work now.  Let me know if thats not the case.   Thanks again for all your help.
